### PR TITLE
RUMM-1174 Fix the integration tests

### PR DESCRIPTION
**Pull request by mariusc83**
### What does this PR do?

This PR addresses following things:

1. We had a bug in the `RumViewScope` that could not close the View when the `ApplicationStart` event was sent during the View lifetime.
2. Due to the new configuration defaults we were sending lots of `LongTask` events during our integration test cases and those were all failing. I had to add a very long threshold for the `LongTaskStrategy` to avoid that and I will open a new Jira task where we specifically add an Espresso test for it.
3. The `integration tests` project was using the `Proguard` based deprecated compiler and not R8.

### Motivation

What inspired you to submit this pull request?

### Additional Notes

Anything else we should know when reviewing?

### Review checklist (to be filled by reviewers)

- [ ] Feature or bugfix MUST have appropriate tests (unit, integration, e2e)
- [ ] Make sure you discussed the feature or bugfix with the maintaining team in an Issue
- [ ] Make sure each commit and the PR mention the Issue number (cf the [CONTRIBUTING](CONTRIBUTING.md) doc)

### CODE REVIEW

File: tools/unit/src/main/kotlin/com/datadog/tools/unit/ObjectTest.kt [hunk 1]
```diff
@@ -28,7 +28,7 @@ abstract class ObjectTest<T : Any> {
      * x.equals(x) is always true
      */
     @Test
-    fun `equals is reflexive`(forge: Forge) {
+    fun equalsIsreflexive(forge: Forge) {
```
Comment (0xnm) [inline tools/unit/src/main/kotlin/com/datadog/tools/unit/ObjectTest.kt:31]: typo: should be `equalsIsreflexive `

File: tools/unit/src/main/kotlin/com/datadog/tools/unit/ObjectTest.kt [hunk 2]
```diff
@@ -63,7 +63,7 @@ abstract class ObjectTest<T : Any> {
      * if (x.equals(y) && y.equals(z)) then x.equals(z)
      */
     @Test
-    fun `equals is transitive`(forge: Forge) {
+    fun equalsIstransitive(forge: Forge) {
```
Comment (0xnm) [inline tools/unit/src/main/kotlin/com/datadog/tools/unit/ObjectTest.kt:66]: typo: not capitalized part

File: dd-sdk-android/src/test/kotlin/com/datadog/android/rum/internal/domain/scope/RumViewScopeTest.kt [hunk 3]
```diff
@@ -1352,7 +1403,7 @@ internal class RumViewScopeTest {
     }
 
     @Test
-    fun `𝕄 send event 𝕎 handleEvent(ApplicationStarted) on stopped view`(
+    fun `𝕄 close the scope 𝕎 handleEvent(ActionSent) on stopped view { ApplicationStarted }`(
```
Comment (0xnm) [inline dd-sdk-android/src/test/kotlin/com/datadog/android/rum/internal/domain/scope/RumViewScopeTest.kt:1406]: just a question: when this (`ApplicationStarted + stopped view`) situation happens?

File: instrumented/integration/src/main/kotlin/com/datadog/android/sdk/integration/RuntimeConfig.kt [hunk 4]
```diff
@@ -28,6 +29,8 @@ internal object RuntimeConfig {
     var tracesEndpointUrl: String = LOCALHOST
     var rumEndpointUrl: String = LOCALHOST
 
+    val LONG_TASK_LARGE_THRESHOLD = TimeUnit.SECONDS.toMillis(240)
```
Comment (0xnm) [inline instrumented/integration/src/main/kotlin/com/datadog/android/sdk/integration/RuntimeConfig.kt:32]: maybe we can use just `Long.MAX_VALUE`?

**Review by 0xnm — CHANGES_REQUESTED** [review]

File: dd-sdk-android/src/test/kotlin/com/datadog/android/rum/internal/domain/scope/RumViewScopeTest.kt (hunk 3 above)
Comment (xgouchet) [reply dd-sdk-android/src/test/kotlin/com/datadog/android/rum/internal/domain/scope/RumViewScopeTest.kt:1406]: I agree, the application started cannot happen on a stopped view in theory… Then again, the reason why is outside of the scope of this class, so this test kinda makes sense

File: tools/unit/src/main/kotlin/com/datadog/tools/unit/ObjectTest.kt (hunk 1 above)
Comment (xgouchet) [inline tools/unit/src/main/kotlin/com/datadog/tools/unit/ObjectTest.kt:31]: Why did you need to rename those methods ?

**Review by xgouchet — APPROVED** [review]

File: tools/unit/src/main/kotlin/com/datadog/tools/unit/ObjectTest.kt (hunk 1 above)
Comment (mariusc83) [reply tools/unit/src/main/kotlin/com/datadog/tools/unit/ObjectTest.kt:31]: We discussed on Slack about this. Is because this module is used by `integrationTests` module and the compiler doesn't understand this syntax.

File: instrumented/integration/src/main/kotlin/com/datadog/android/sdk/integration/RuntimeConfig.kt (hunk 4 above)
Comment (mariusc83) [reply instrumented/integration/src/main/kotlin/com/datadog/android/sdk/integration/RuntimeConfig.kt:32]: Oh yes...that makes sense.

File: dd-sdk-android/src/test/kotlin/com/datadog/android/rum/internal/domain/scope/RumViewScopeTest.kt (hunk 3 above)
Comment (mariusc83) [reply dd-sdk-android/src/test/kotlin/com/datadog/android/rum/internal/domain/scope/RumViewScopeTest.kt:1406]: Like @xgouchet explained below, this is just to make sure we test all the scenarios in this class.

File: tools/unit/src/main/kotlin/com/datadog/tools/unit/ObjectTest.kt (hunk 1 above)
Comment (0xnm) [reply tools/unit/src/main/kotlin/com/datadog/tools/unit/ObjectTest.kt:31]: per this https://stackoverflow.com/a/62597742 backticked methods should work with settings we have, but people still report issues with 4.1.2, so it is better to use non-backticked names format imo. indeed it is (was) DEX limitation.

**Review by 0xnm — APPROVED** [review]
